docs/api/automod.rst

```diff
@@ -0,0 +1,166 @@
+.. SPDX-License-Identifier: MIT
+
+.. currentmodule:: disnake
+
+AutoMod
+=======
+
+This section documents everything related to guilds' AutoMod feature.
```

Comment:
A link to https://discord.com/developers/docs/resources/auto-moderation would be nice

Reply:
Done in [`d15ddd7`](https://github.com/DisnakeDev/disnake/pull/722/commits/d15ddd70e20c611070a3445379763a80b07a8c13)